test/integration/kubernetes_deploy_test.rb

```diff
@@ -1,27 +1,128 @@
 require 'test_helper'
 
 class KubernetesDeployTest < KubernetesDeploy::IntegrationTest
-  def test_basic
-    runner = KubernetesDeploy::Runner.new(
-      namespace: @namespace,
-      current_sha: "abcabcabc",
-      context: MINIKUBE_CONTEXT,
-      template_dir: File.expand_path("./test/fixtures/basic"),
-      wait_for_completion: true,
-    )
-    runner.run
-
-    pods = kubeclient.get_pods(namespace: @namespace)
-    managed_pods, unmanaged_pods = pods.partition do |pod|
-      pod.metadata.ownerReferences && pod.metadata.ownerReferences.first.kind == "ReplicaSet"
+  def test_full_basic_set_deploy_succeeds
+    deploy_fixtures("basic")
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_up
+  end
+
+  def test_partial_deploy_followed_by_full_deploy
+    deploy_fixtures("basic", subset: ["configmap-data.yml", "redis.yml"])
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_redis_resources_up
+    basic.assert_configmap_data_present
+    basic.refute_managed_pod_exists
+    basic.refute_web_resources_exist
+
+    deploy_fixtures("basic")
+    basic.assert_all_up
+  end
+
+  def test_pruning
+    deploy_fixtures("basic")
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_up
+
+    deploy_fixtures("basic", subset: ["redis.yml"])
+    basic.assert_all_redis_resources_up
+    basic.refute_configmap_data_exists
+    basic.refute_managed_pod_exists
+    basic.refute_web_resources_exist
+  end
+
+  def test_pvcs_are_not_pruned
+    deploy_fixtures("basic", subset: ["redis.yml"])
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_redis_resources_up
+
+    deploy_fixtures("basic", subset: ["configmap-data.yml"])
+    basic.assert_configmap_data_present
+    basic.refute_redis_resources_exist(expect_pvc: true)
+  end
+
+  def test_success_with_unrecognized_resource_type
+    # Secrets are intentionally unsupported because they should not be committed to your repo
+    secret = {
+      "apiVersion" => "v1",
+      "kind" => "Secret",
+      "metadata" => { "name" => "test" },
+      "data" => { "foo" => "YmFy" }
+    }
+
+    deploy_fixtures("basic", subset: ["configmap-data.yml"]) do |fixtures|
+      fixtures["secret.yml"] = { "Secret" => secret }
     end
 
-    assert_equal 2, managed_pods.size
-    managed_pods.each do |pod|
-      assert_equal "Running", pod.status.phase
+    live_secret = kubeclient.get_secret("test", @namespace)
+    assert_equal({ foo: "YmFy" }, live_secret["data"].to_h)
+  end
+
+  def test_invalid_yaml_fails_fast
+    assert_raises(KubernetesDeploy::FatalDeploymentError, /Template \S+yaml-error\S+ cannot be parsed/) do
+      deploy_dir(fixture_path("invalid"))
```

Comment:
The purpose of this test is to cover deploy behaviour in the case where invalid yaml is provided. It needs circumvent the loading of the yaml file into a ruby structure, or else the yaml parse exception will be raised in the setup helpers instead of during the deploy. So we can't get it to use `deploy_fixtures`, but I can have it explicitly use the `KubernetesDeploy::Runner` instead of exposing the helper if you prefer.